widgets/src/loading_spinner.rs

```diff
@@ -6,68 +6,49 @@ live_design!{
-
-                // Direction factor: >0 increases, <0 decreases
-                let direction = sign(sin(theta));
-                let arc_length = (1.0 - cos(theta)) * max_arc / 2.0;
-                let signed_arc_length = direction * arc_length;
-
-                // Defining the start and end points of an arc
-                let start_angle = tail_angle + min(signed_arc_length, 0.0);
-                let end_angle = tail_angle + max(signed_arc_length, 0.0);
-
-                // Calculate the angle of the pixel relative to the starting point
-                let relative_angle = mod(angle - start_angle, 2.0 * PI);
-                let arc_length_abs = abs(signed_arc_length);
-                let arc_alpha = float(relative_angle < arc_length_abs);
-
-                // Calculate the angle of the pixel relative to the starting point
-                let cap_center_tail = vec2(cos(start_angle), sin(start_angle)) * ring_center;
-                let cap_center_head = vec2(cos(end_angle), sin(end_angle)) * ring_center;
-
-                // Calculate the distance from the pixel to the center of the semicircle
-                let dist_to_tail_cap = length(pos - cap_center_tail);
-                let dist_to_head_cap = length(pos - cap_center_head);
-
-                // Determine if the pixel is within the head or tail semicircle
-                let in_tail_cap = dist_to_tail_cap < cap_radius;
-                let in_head_cap = dist_to_head_cap < cap_radius;
-
-                // Transparency of merged arcs and semicircles
-                let alpha = clamp((arc_alpha + float(in_tail_cap) + float(in_head_cap)) * d, 0.0, 1.0);
-
-                return vec4(self.color.rgb * alpha, alpha);
+                let sdf = Sdf2d::viewport(self.pos * self.rect_size);
+                
+                let center = self.rect_size * 0.5;
+                
+                let rotation = self.time * self.rotation_speed * 2.0 * PI;
+                
+                let rotation_cycles = rotation / (2.0 * PI);
+                let arc_phase = mod(rotation_cycles * 0.5, 1.0);
+                
+                let expand_phase = clamp(arc_phase / 0.55, 0.0, 1.0);
+                let contract_phase = clamp((arc_phase - 0.55) / 0.45, 0.0, 1.0);
+                
+                let cycle = expand_phase * (1.0 - contract_phase);
+                
+                let gap_ratio = mix(self.min_gap_ratio, self.max_gap_ratio, cycle);
+                let gap_radians = gap_ratio * 2.0 * PI;
+                
+                let start_angle = rotation;
+                
+                sdf.arc_round_caps(
+                    center.x, 
+                    center.y, 
+                    self.radius, 
+                    start_angle, 
+                    start_angle + 2.0 * PI - gap_radians, 
+                    self.stroke_width
+                );
+                
+                return sdf.fill(vec4(self.spinner_color, 1.0));
```

Comment:
see my above comment.

```suggestion
                return sdf.fill(self.spinner_color);
```